controllers/openstackbaremetalset_controller.go

```diff
@@ -148,10 +145,9 @@ func (r *OpenStackBaremetalSetReconciler) Reconcile(ctx context.Context, req ctr
 
 		if statusChanged() {
 			if updateErr := r.Client.Status().Update(context.Background(), instance); updateErr != nil {
+				err = common.WrapErrorForObject(
+					"Update Status", instance, updateErr)
 				if err == nil {
```

Comment:
Will `err` ever be `nil` here?  Since `updateErr` is not `nil` and gets passed to `WrapErrorForObject`, it will be return by that func as non-`nil` `err`.

Reply:
right that should be != nil

Reply:
or just use LogErrorForObject